internal/pipe/ko/ko.go

```diff
@@ -125,11 +127,16 @@ func (Pipe) Default(ctx *context.Context) error {
 			ko.SBOM = "spdx"
 		}
 
+		if ko.Repository != "" {
+			ko.Repositories = append(ko.Repositories, ko.Repository)
+			deprecate.Notice(ctx, "kos.repository")
+		}
+
 		if repo := ctx.Env["KO_DOCKER_REPO"]; repo != "" {
-			ko.Repository = repo
+			ko.Repositories = []string{repo}
 		}
 
-		if ko.Repository == "" {
+		if len(ko.Repositories) == 0 {
 			return errNoRepository
```

Comment:
I guess, we should change also the message text for `errNoRepository`. Currently, it mentions `repository` field:

```go
errNoRepository = errors.New("ko: missing repository: please set either the repository field or a $KO_DOCKER_REPO environment variable")
```

Reply:
good call! thanks!